### ISSUE TITLE
[enhancement] train.py 구현

### ISSUE THREAD

**Issue by hanbyeol1001** [boostcampaitech7/level2-cv-semanticsegmentation-cv-22-lv3#13]
# **추가된 기능에 대한 설명을 작성하세요**
- [x] 모듈화
- [x] config file
- [x] debugging mode (cfg.debug=True 일 경우 실험 환경이 잘 설정되었는지 체크하기 위해 epoch을 1~2로 설정하여 진행되는 모드)
<br><br>

# **수정해야 할 파일들**
- [x] trainer.py 추가
- [x] train.py 추가
- [x] dataset 관련 py파일(made by. JSM) 위치 변경
<br><br>

# **추가 전달 사항**
추가적으로 구현하면 좋을 것
- Optuna 사용(하이퍼파라미터 튜닝 중에서 도움이 되는 라이브러리)
-  wandb setting
- validation set test 과정에서 dice 계산 cpu -> gpu
- debugging mode (cfg.debug=True 일 경우 실험 환경이 잘 설정되었는지 체크하기 위해 epoch을 1~2로 설정하여 진행되는 모드)
- 여러 loss function 중 선택
- loss 출력하는것도 config로 하도록 수정하는것도 좋아보인다.
- 내가 원하는 epoch 별 체크포인트 저장하는 코드 추가
